test/add.js

```diff
@@ -38,222 +38,268 @@ function makeFileWithInlineSourceMap() {
 		contents: new Buffer('console.log("line 1.1"),console.log("line 1.2"),console.log("line 2.1"),console.log("line 2.2");\n//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYWxsLmpzIiwic291cmNlcyI6WyJ0ZXN0MS5qcyIsInRlc3QyLmpzIl0sIm5hbWVzIjpbImNvbnNvbGUiLCJsb2ciXSwibWFwcGluZ3MiOiJBQUFBQSxRQUFBQyxJQUFBLFlBQ0FELFFBQUFDLElBQUEsWUNEQUQsUUFBQUMsSUFBQSxZQUNBRCxRQUFBQyxJQUFBIiwic291cmNlc0NvbnRlbnQiOlsiY29uc29sZS5sb2coJ2xpbmUgMS4xJyk7XG5jb25zb2xlLmxvZygnbGluZSAxLjInKTtcbiIsImNvbnNvbGUubG9nKCdsaW5lIDIuMScpO1xuY29uc29sZS5sb2coJ2xpbmUgMi4yJyk7Il0sInNvdXJjZVJvb3QiOiIvc291cmNlLyJ9')
 	});
 }
+describe('add.js', function() {
 
-test('add: should return an error when no valid vinyl file is provided', function (t) {
-	[
-		{ type: 'null', val: null },
-		{ type: 'an object', val: {} },
-		{ type: 'a stream', val: new stream.Readable() }
-	].map(function(obj) {
-		sourcemaps.add(obj.val, function(err, file) {
-			t.ok(err instanceof Error && err.message === 'vinyl-sourcemap-add: Not a vinyl file', util.format('should not accept %s as first argument', obj.type));
+	it('should not accept null as argument', function(done) {
+		[
+			{type: 'null'}
+		].map(function(obj) {
+			sourcemaps.add(obj.val, function(err, file) {
+				expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Not a vinyl file').toExist();
+				done();
+			});
 		});
 	});
-	t.end();
-});
 
-test('add: should return an error when invalid options are provided', function (t) {
-	var file = makeFile();
-	[
-		{ type: 'undefined' },
-		{ type: 'null', val: null },
-		{ type: 'a string', val: '' },
-		{ type: 'a boolean', val: true }
-	].map(function(obj) {
-		sourcemaps.add(file, obj.val, function(err, file) {
-			t.ok(err instanceof Error && err.message === 'vinyl-sourcemap-add: Invalid argument: options', util.format('should not accept %s as options argument', obj.type));
+
+
+	it('should not accept an empty object as argument', function(done) {
+		[
```

Comment:
No need to use an array and `.map` here. You can just replace `obj.val` with the value you are testing